### ISSUE TITLE
[Bug]: Narrator Support Issue - Electron App does not emit property/structure changed event

### ISSUE THREAD

**Issue by blueset** [electron/electron#43520]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

32.0.1

### What operating system(s) are you using?

Windows

### Operating System Version

Windows 11 build 27686

### What arch are you using?

x64

### Expected Behavior

In an Electron app, Narrator should be able to announce to the user when a checkbox is toggled, an element is expanded/collapsed or when a button is pressed.

### Actual Behavior

Narrator is able to announce elements when they gain focus, but fails to announce any state changes such as the ones listed in the expected behavior.

### Testcase Gist URL

https://gist.github.com/mliao95/acb071de0837118c83586a7adb0d21f3

### Additional Information

This is a reproduction of #28479.
